projects/gui/src/mainwindow.cpp

```diff
@@ -564,6 +564,16 @@ void MainWindow::setCurrentGame(const TabData& gameData)
 		m_evalWidgets[i]->setPlayer(player);
 	}
 
+	// if a human plays the black side against an engine then flip the board
+	if (m_players[0]->isHuman() != m_players[1]->isHuman())
```

Comment:
I don't think it makes sense to loop over the players when we know that the white player is always `m_players[0]` or `m_players[Chess::Side::White]` and black player is `m_players[1]` or `m_players[Chess::Side::Black]`. I think this should be all that is needed:

```
if (!m_players[Chess::Side::White]->isHuman()
&&  m_players[Chess::Side::Black]->isHuman())
	m_gameViewer->boardScene()->flip();
```

Also, as a side note, taking a reference to a pointer type (`const auto& player`) doesn't usually make sense.